edalize/templates/libero/libero-project.tcl.j2

```diff
@@ -0,0 +1,29 @@
+# This file is generated by Edalize.
+# Microsemi Tcl Script
+# Libero
+
+new_project -location {{op}}{{prj_root}}{{cl}} -name {{name}} -project_description {} -block_mode 0 -standalone_peripheral_initialization 0 -instantiate_in_smartdesign 1 -ondemand_build_dh 1 -hdl {{op}}{{tool_options.hdl}}{{cl}} -family {{op}}{{tool_options.family}}{{cl}} -die {{op}}{{tool_options.die}}{{cl}} -package {{op}}{{tool_options.package}}{{cl}} -speed {{op}}{{tool_options.speed}}{{cl}} -die_voltage {{op}}{{tool_options.dievoltage}}{{cl}} -part_range {{op}}{{tool_options.range}}{{cl}} -adv_options {IO_DEFT_STD:{{tool_options.defiostd}}{{cl}} -adv_options {RESTRICTPROBEPINS:1} -adv_options {RESTRICTSPIPINS:0} -adv_options {SYSTEM_CONTROLLER_SUSPEND_MODE:0} -adv_options {TEMPR:{{tool_options.range}}{{cl}} -adv_options {VCCI_1.2_VOLTR:EXT} -adv_options {VCCI_1.5_VOLTR:EXT} -adv_options {VCCI_1.8_VOLTR:EXT} -adv_options {VCCI_2.5_VOLTR:EXT} -adv_options {VCCI_3.3_VOLTR:EXT} -adv_options {VOLTR:{{tool_options.range}}{{cl}}
+
+set_device -family {{op}}{{tool_options.family}}{{cl}} -die {{op}}{{tool_options.die}}{{cl}} -package {{op}}{{tool_options.package}}{{cl}} -speed {{op}}{{tool_options.speed}}{{cl}} -die_voltage {{op}}{{tool_options.dievoltage}}{{cl}} -part_range {{op}}{{tool_options.range}}{{cl}} -adv_options {IO_DEFT_STD:{{tool_options.defiostd}}{{cl}} -adv_options {RESTRICTPROBEPINS:1} -adv_options {RESTRICTSPIPINS:0} -adv_options {SYSTEM_CONTROLLER_SUSPEND_MODE:0} -adv_options {TEMPR:{{tool_options.range}}{{cl}} -adv_options {VCCI_1.2_VOLTR:EXT} -adv_options {VCCI_1.5_VOLTR:EXT} -adv_options {VCCI_1.8_VOLTR:EXT} -adv_options {VCCI_2.5_VOLTR:EXT} -adv_options {VCCI_3.3_VOLTR:EXT} -adv_options {VOLTR:{{tool_options.range}}{{cl}}
+
+# Link directly to source files
+create_links \
+         -convert_EDN_to_HDL 0 \
```

Comment:
I haven't worked with Libero in ten years and I find the documentation a bit confusing. The hdl_source_folder is not documented and I can't really find what the difference is between import_files and create_links

But with that said we can't import a directory like that. In many cases it's totally fine to have an undefined order between files, but FuseSoC and Edalize keeps guarantees on the file order because there are many cases where that matters. So we need to list all files. Question is if create_links or import_files should be used.

We can drop the `convert_EDN_to_HDL` option as that is only valid for importing EDIF files as I understand it

We also need to set the correct library (at least for VHDL files) instead of hardcoding work. The library name can be found in the `logical_name` property of files

Reply:
I was using import_files before but some files didn't get imported like a `.vh` file from corescore.. will investigate this further.
Also I understand that Libero puts all user files under `work` library. Should we use from logical_name?

Reply:
Also when changing the logic to import the files, I don't see files marked as "is_include_file" like "rtl/serv_params.vh" from serv being listed on src_files.

How this should be handled?

Reply:
Verilog include files work pretty similar to C include files so instead of putting them in the list of files to compile,  we just make sure that the directories of these files can be found. The `get_fileset_files` helper returns a list of files that are not include files and a list of directories where include files are found. The problem here is that I can't find a way in Libero to set include directories so I'm not really sure what to do yet. I need to investigate this further

And regarding library/logical_name, `work` is the default name, so we can use that unless the user has explicitly set a logical_name

Reply:
I realized a similar problem is how to set top-level parameters and defines which are things we need to do as well. I'm starting to suspect we need to look in to the `configure_tool` command and send all these things directly to synplify_pro, which is the tool that handles the actual synthesis inside of Libero. I am not sure yet though

Reply:
Fixed the import paths passing it thru a TCL script to Symplify by using a Libero config.
Thanks for you pointing that out.